lib/Widgets/DynamicNotebook.vala

```diff
@@ -1179,18 +1189,25 @@ namespace Granite.Widgets {
             }
         }
 
-        private void on_close_others (Tab tab) {
-            var num = 0; //save num, in case a tab refused to close so we don't end up in an infinite loop
-
-            for (var j = 0; j < tabs.length (); j++) {
-                if (tab != tabs.nth_data (j)) {
-                    tabs.nth_data (j).closed ();
-                    if (num == n_tabs) break;
-                    j--;
+        private void on_close_others (Tab clicked_tab) {
+            tabs.copy ().foreach ((tab) => {
```

Comment:
It looks like `tabs` is recreated every time the property is being accessed: https://github.com/andreasfelix/granite/blob/close_tabs_to_the_right/lib/Widgets/DynamicNotebook.vala#L679-L688

I'm not sure if `copy` is necessary or wasteful in this situation. I also don't quite understand why there's a `_tabs` field to hold the value if it's being recreated anyway :thinking:

Reply:
Maybe the reasoning is similar to this temporary (seven year old :grimacing:) field?

https://github.com/andreasfelix/granite/blob/close_tabs_to_the_right/lib/Widgets/DynamicNotebook.vala#L671

Reply:
I added the `copy`, because otherwise I get a segmentation fault.

Reply:
Ah right, because returned properties are not owned, and the next "call" to `tabs` will destroy the old list you're iterating over. All good then :+1: